api/application/client.go

```diff
@@ -102,6 +107,16 @@ type DeployArgs struct {
 // it. Placement directives, if provided, specify the machine on which the charm
 // is deployed.
 func (c *Client) Deploy(args DeployArgs) error {
+	if len(args.AttachStorage) > 0 && args.NumUnits != 1 {
+		return errors.Errorf("AttachStorage is non-empty, but NumUnits is %d", args.NumUnits)
```

Comment:
Done. The error message is intended for the programmer that writes to the API, but I guess we should assume that they'll just send the error up to the user.